### PR TITLE
Update to use MyQ API v5

### LINKED ISSUE #44: Error getting MyQ devices: undefined
Was working yesterday.

[7/1/2020, 13:33:20] [MyQ2] Error getting MyQ devices: undefined
[7/1/2020, 13:33:36] [MyQ2] Error getting MyQ devices: undefined
[7/1/2020, 13:33:44] [Config] Starting terminal session

### PR DESCRIPTION
**Pull request by shamoon**
Closes #44, #45, #46, #48

### CODE REVIEW

**Comment by ryanschmidt** [comment]
@shamoon Thanks for this. I tested it and I'm getting `Unable to login to MyQ: undefined` and the devices do not show in HomeKit. All was working before the recent API changed.

**Comment by shamoon** [comment]
Wow looks like they changed something again. Hold please...

**Comment by shamoon** [comment]
@ryanschmidt please try it now. FYI, the repo owner has said he wont be merging this because he's working on a larger revamp that should be ready soon.

In the meantime, if youre still having issues with this branch (same to you @Gilmargolin) enable `debug` in the config and paste your logs here (remove any email/pw etc.).

**Comment by ryanschmidt** [comment]
@shamoon I really appreciate the help! I'm getting `Error: Couldn't find a MyQ door device.` and not sure if you mean `verbose` logging for `myq2` or debug for homebridge. Let me know and I can definitely share that.

**Comment by shamoon** [comment]
I mean this plugin’s debug option. If you’re using the newest version of this plugin (with my code in this branch in index.js) then it should be `debug` not `verbose`, but yea that one should give you lots more output.

**Comment by ryanschmidt** [comment]
@shamoon Here's the output

```
[7/3/2020, 12:29:38 AM] [MyQ2] Error: Couldn't find a MyQ door device.
[7/3/2020, 12:29:47 AM] [MyQ2] Unable to login to MyQ, received error: FetchError: request to https://api.myqdevice.com/api/v5/Login failed, reason: read ECONNRESET
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/homebridge-myq2/node_modules/node-fetch/lib/index.js:1455:11)
    at ClientRequest.emit (events.js:210:5)
    at TLSSocket.socketErrorListener (_http_client.js:406:9)
    at TLSSocket.emit (events.js:210:5)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  message: 'request to https://api.myqdevice.com/api/v5/Login failed, reason: read ECONNRESET',
  type: 'system',
  errno: 'ECONNRESET',
  code: 'ECONNRESET'
}
[7/3/2020, 12:29:52 AM] [MyQ2] Error: Couldn't find a MyQ door device.
[7/3/2020, 12:29:58 AM] [MyQ2] Error: Couldn't find a MyQ door device.
```

**Comment by shamoon** [comment]
Oh yea thats like youre not even able to reach the MyQ server. I'm not sure if your IP is rate-limited (I actually have another PR thats also not getting merged that slows down login attempts after stuff like this 😃). Are you able to load that URL in a browser (you could test in Postman to see if you can login but loading in a browser should at least tell us if youre rate-limited or something like that)?

**Comment by nixta** [comment]
@shamoon Huge thanks for pulling this together. I just hand-merged (😬) this and your rate limiting PR into my source and it's working a treat.

File: index.js [hunk 1]
```diff
@@ -273,83 +301,47 @@ MyQ2Platform.prototype.getDevice = function (callback) {
...
       for (var i = 0; i < devices.length; i++) {
         var device = devices[i];
-        var deviceType = device.MyQDeviceTypeName;
+        var deviceType = device.device_type;
 
         // Search for specific device type
-        if(deviceType === "Garage Door Opener WGDO" || deviceType === "GarageDoorOpener" || deviceType === "VGDO" || deviceType === "Gate") {
-          var thisDeviceID = device.MyQDeviceId.toString();
-          var thisSerial = device.SerialNumber.toString();
+        if(deviceType == 'virtualgaragedooropener' || deviceType === "Garage Door Opener WGDO" || deviceType === "GarageDoorOpener" || deviceType === "VGDO" || deviceType === "Gate") {
```
Comment (mstansberry) [inline index.js:318]: when testing, had to add `|| deviceType == "garagedooropener"` to get this working with recently added liftmaster

error prior to adding:
`[7/3/2020, 5:07:18 PM] [MyQ2] Error: Couldn't find a MyQ door device.`

**Comment by Moving-Electrons** [comment]
Hi, I installed the last version of the plugin (through the Homebridge Config UI X) and now I'm getting the following error:
_[MyQ] Error getting MyQ devices: undefined_

File: index.js (hunk 1 above)
Comment (shamoon) [reply index.js]: Cheers, I added that. Yea I only have one device to test with =)

**Comment by shamoon** [comment]
@Moving-Electrons Please see the comment above: https://github.com/hjdhjd/homebridge-myq2/pull/47#issuecomment-653335227 . This PR has not been merged by the repo owner, thus is not included when you install via Config UI X, you would have to apply this change manually.

**Comment by Moving-Electrons** [comment]
> @Moving-Electrons Please see the comment above: [#47 (comment)](https://github.com/hjdhjd/homebridge-myq2/pull/47#issuecomment-653335227) . This PR has not been merged by the repo owner, thus is not included when you install via Config UI X, you would have to apply this change manually.

Thanks @shamoon . How can I install this branch manually?. Also, would it be automatically replaced when this PR is merged ?.

**Comment by zabernoki** [comment]
Thanks for all the work on this @shamoon. I've got things working and a device is showing up in HomeKit. However, I have two hubs and it looks like it will only add devices from the first hub it finds. It's adding the Garage Door fine but won't grab the Gate. Any thoughts on what might be happening here? 

`    {
      device_family: 'garagedoor',
      device_platform: 'myq',
      device_type: 'virtualgaragedooropener',
      name: 'Garage Door',
      created_date: '2019-12-10T22:40:25.807',
      state: [Object]
    },
    {
      device_family: 'garagedoor',
      device_platform: 'myq',
      device_type: 'gate',
      name: 'Gate',
      created_date: '2020-07-03T18:47:30.387',
      state: [Object]
    },
    {
      device_family: 'gateway',
      device_platform: 'myq',
      device_type: 'hub',
      name: 'Hub 2',
      created_date: '2019-12-10T22:39:36.367',
      state: [Object]
    },
    {
      device_family: 'gateway',
      device_platform: 'myq',
      device_type: 'ethernetgateway',
      name: 'LiftMaster Gateway',
      created_date: '2019-02-09T17:39:17.947',
      state: [Object]
    }`

** Removed serials and api urls since I wasn't sure what is considered a unique identifier. **

**Comment by shamoon** [comment]
@zabernoki Yea this hasn't really been tested much since theres kinda no point if it will never get merged =) but please try it again, I added code to allow for `gate` devices

**Comment by zabernoki** [comment]
Well now I feel dumb because I didn't check for capitalization. Could have saved you some time. *facepalm* 
Appreciate you stepping in while we wait for the bigger update.

**Comment by shamoon** [comment]
@Moving-Electrons barring any inadvertent permissions problems any changes you make will be overwritten when the actual plugin is updated (worst case you'd just remove your edited version and re-install). As for how to manually change it, easiest thing is to copy / paste, or you can use git command line, etc. Sorta depends how comfortable you are with this stuff... see https://www.reddit.com/r/homebridge/comments/hjz9hh/error_getting_myq_devices_unauthorized_216/fwtnnik/

File: index.js (hunk 1 above)
Comment (ryanschmidt) [reply index.js]: I'm still getting `Error: Couldn't find a MyQ door device.` so there must be another deviceType that we need. I'll check this out tomorrow and grab the name of my device so we can add it here. Thanks again!

Comment (shamoon) [reply index.js]: Hopefully by tomorrow that wont be necessary one way or the other =)

FYI if you enable `debug` logging on this plugin there would be some log output that might be helpful to figure out your issue.

**Comment by hjdhjd** [comment]
Closing this and deprecating.

**Pull request closed without merging**